pkg/queue/postgres/setup.go

```diff
@@ -0,0 +1,102 @@
+package postgres
+
+import (
+	"bytes"
+	"context"
+	"text/template"
+
+	"github.com/contiamo/go-base/pkg/db"
+)
+
+type ForeignReference struct {
+	ColumnName       string
+	ColumnType       string
+	ReferencedTable  string
+	ReferencedColumn string
+}
+
+func Setup(ctx context.Context, db db.SQLDB, references []ForeignReference) error {
```

Comment:
We need a test that makes sure this generates valid SQL.

Reply:
I added a full usage test on the ForeignReference argument, including a cascading delete of a task that executes because of it.